coins/src/adapters/lps/uniswap/uniswap.ts

```diff
@@ -1,83 +1,210 @@
-import { PromisePool } from '@supercharge/promise-pool'
-
-const pairsPerCall = 100
-const lpQuery = gql`
-query lps($firstLp: ID!){
-	pairs(first:${pairsPerCall}, where:{id_gte: $firstLp}){
-		id
-		reserve0,
-		reserve1,
-		totalSupply,
-		token0{
-			id
-		},
-		token1{
-			id
-		},
-	}
-}
-`
-
-const simulatenousCalls = 1000;
-export async function getPairPrices(chain: string, factory: string, subgraph: string) {
-    const pairsLength = Number((await call({
-        target: factory,
-        chain: chain as any,
-        abi: abi.allPairsLength
-    })).output)
-    const calls = Array.from(Array(Math.floor(pairsLength / pairsPerCall)).keys()).map(i => ({
-        target: factory,
-        params: i
+import { multiCall, call } from "@defillama/sdk/build/abi/index";
+import abi from "./abi.json";
+import {
+  addToDBWritesList,
+  getTokenAndRedirectData
+} from "../../utils/database";
+import { getLPInfo } from "../../utils/erc20";
+import { write, read } from "../../utils/dbInterfaces";
+import { multiCallResults } from "../../utils/sdkInterfaces";
+import { requery } from "../../utils/sdk";
+import { request, gql } from "graphql-request";
+
+async function fetchUniV2Markets(chain: string, factory: string) {
+  let pairsLength: string = (
+    await call({
+      target: factory,
+      chain: chain as any,
+      abi: abi.allPairsLength
+    })
+  ).output;
+
+  const pairNums: number[] = Array.from(Array(Number(pairsLength)).keys());
+
+  const pairs: multiCallResults = await multiCall({
+    abi: abi.allPairs,
+    chain: chain as any,
+    calls: pairNums.map((num) => ({
+      target: factory,
+      params: [num]
     }))
-    const lpsEveryStep = await multiCall({
-        target: factory,
-        chain: chain as any,
-        calls,
-        abi: abi.allPairs,
-        requery: true,
+  });
+
+  await requery(pairs, chain, abi.allPairs);
+  return pairs.output.map((result) => result.output.toLowerCase());
+}
+async function fetchUniV2MarketsFromSubgraph(subgraph: string) {
+  let addresses: string[] = [];
+  let reservereThreshold: number = 0;
+  for (let i = 0; i < 5; i++) {
+    const lpQuery = gql`
+      query lps {
+        pairs(first: 1000, orderBy: reserveUSD, orderDirection: desc ${
+          i == 0
+            ? ``
+            : `,         
+          where: {
+            reserveUSD_lt: ${reservereThreshold}
+          }`
+        }) {
+          id
+          reserveUSD
+        }
+      }`;
+    const result = (await request(subgraph, lpQuery)).pairs;
+    reservereThreshold = result[999].reserveUSD;
+    addresses.push(
+      ...(await request(subgraph, lpQuery)).pairs.map((p: any) => p.id)
+    );
+  }
+  return addresses;
+}
+async function fetchUniV2MarketData(chain: string, pairAddresses: string[]) {
+  let token0s: multiCallResults;
+  let token1s: multiCallResults;
+  let reserves: multiCallResults;
+  [token0s, token1s, reserves] = await Promise.all([
+    multiCall({
```

Comment:
what about setting requery:true, instead of calling requery afterwards?

in this case we can do that since we know for sure that all calls must be successful